pylint/checkers/base.py

```diff
@@ -2588,10 +2588,78 @@ def _check_type_x_is_y(self, node, left, operator, right):
         self.add_message("unidiomatic-typecheck", node=node)
 
 
+class ModifiedIterationChecker(_BasicChecker):
+    """Checks for modified for loops iterations
+    Currently supports `for` loops for Sets, Dictionaries and Lists.
+    """
+
+    __implements__ = interfaces.IAstroidChecker
+
+    msgs = {
+        "W4714": (
```

Comment:
Not sure if @Pierre-Sassoulas agrees but I think the errors can go in `BasicErrorChecker`. Although W4714 is not technically an error I think it makes sense to keep them together, so I'd suggest to move it there as well.

Otherwise I think we can move the two error messages to `BasicErrorChecker` and put the warning in `MisdesignChecker`.

That would also invalidate [this comment](https://github.com/PyCQA/pylint/pull/5628/files#r777143397)

Reply:
Sure, it make sense. We're going to burst base.py into file before releasing 2.13 in #5489 anyway so having the file temporarily be bigger because old checker already inside it get bigger is not a problem.

Reply:
I didn't understand what you guys mean.. I am not familiar with those 2 classes...